### PR TITLE
Developer Tools MVP

### LINKED ISSUE #1075: Keyboard shortcut to enable/display the developer tools, and alternatively another way to open them on mobile.
Using this as a starting point for developer tools. I thought starting with a keyboard shortcut is best so that people are not too confused by a new button without much functionality.

### PR DESCRIPTION
**Pull request by bjosttveit**
## Description

This is the starting point for developer tools. This PR adds the developer tools panel and a few simple features to begin with.

**[Related epic](https://github.com/Altinn/app-frontend-react/issues/1021)**

- It can be opened with the hotkey `Ctrl+Shift+K`/`⌘+Shift+K`.
- There is a button hiding away in the bottom right corner that will also open it. (yes I tested it on mobile)
  - This is only visible in `local.altinn.cloud`.
- The panel is resizable (also works on mobile with touch events 😩 )
- **Functionality**:
  - PDF preview button
  - Navigate pages that are in the current page order.

Video of the current capabilities:

https://user-images.githubusercontent.com/47412359/229771638-60ab60aa-02fa-49bf-b76d-1f9f17da3739.mov

## Related Issue(s)

- closes #1075 

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [ ] Cypress E2E test(s) have been added/updated
  - [x] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [ ] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [x] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [ ] No functionality has been changed/added, so no documentation is needed
  - [x] I will do that later/have created an issue
  
- Changes/additions to component properties
  - [ ] Changes are reflected in both `src/layout/layout.d.ts` and `layout.schema.v1.json`, and these are all backwards-compatible
  - [x] No changes made
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [?] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

**Review by olemartinorg — APPROVED** [review]
Great stuff! 👏 🚀 Maybe we could add a message here informing newcomers about what this is, and how to open/close it via the keyboard for tt02/production apps (so they'll get a hint on `local.altinn.cloud`, but knows that the feature is available in other environments as well).

Going forwards, I think the firefox/chrome devtools layout is nice, and something we could copy. By that, I mean less padding, tabs to navigate between different devtools features, etc.

File: src/features/devtools/DevTools.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  children: ReactNode;
+}
+
+export const DevTools = ({ children }: IDevToolsProps) => {
+  const [panelOpen, setPanelOpen] = useState(false);
+
+  useEffect(() => {
+    if (devToolsEnabled) {
+      return window.addEventListener('keydown', (event) => {
+        if ((event.ctrlKey || event.metaKey) && event.key === 'k') {
```
Comment (ivarne) [inline src/features/devtools/DevTools.tsx:19]: Isn't this shortcut rather short and likely for normal users to press inadvertently.

**Comment by bjosttveit** [comment]
Just hide the button in tt02/prod? So you think it should be available in prod as well?

How do you think I should give the information about the availability? a pop-up message?

Totally agree on the layout stuff, I had to stop myself from going down that rabbit hole now since there are only two features currently 😅

**Review by ivarne — COMMENTED** [review]
This seems like a nice place to start using code splitting. If devtools grows to provide half of the suggested features, it would be nice if everyone didn't need to download functionality they won't use.

File: src/features/devtools/DevTools.tsx (hunk 1 above)
Comment (bjosttveit) [reply src/features/devtools/DevTools.tsx:19]: I will gladly take suggestions for other shortcuts, maybe just add `Shift`?

Comment (ivarne) [reply src/features/devtools/DevTools.tsx:19]: I think i missunderstood this. It's currently just a shortcut for when the icon is already visible?

I'm thinking about when this is activated in prod, the mode should probably be activated by something else than a key combination that others might have configured for other purposes (or inadvertently type). Personally I'd suggest setting a custom cookie (devtoolsEnable:true) from javascript when the user adds `?devtools=` in their url. The devtools can provide a button to delete the cookie, and docs entries can link to any app to help users set the cookie. 

PS: I don't think code splitting should be dependent on `Index.cshtml`, but I have not tried setting it up myself. I just used Create React App.

**Comment by bjosttveit** [comment]
A message like this? @olemartinorg 

image

File: src/features/devtools/DevTools.tsx (hunk 1 above)
Comment (olemartinorg) [reply src/features/devtools/DevTools.tsx]: I believe we'll actually want to have the devtools somewhat easily accessible in prod as well. The goal here is not to hide anything, even from competent users (they could figure all of this out without devtools if they wanted), and using a cookie or custom URL just adds more friction than we really need. My goal is to make these tools available on mobile as well some day, and making mobile users inject a cookie will make it very hard for them to use the tools. (Keyboard shortcuts isn't all too easy on mobile either, but we'll figure something out in time.)

**Pull request merged**